kloppy/domain/services/transformers/dataset.py

```diff
@@ -431,3 +435,53 @@ def transform_dataset(
             )
         else:
             raise KloppyError("Unknown Dataset type")
+
+
+class DatasetTransformerBuilder:
+    def __init__(
+        self, to_coordinate_system: Optional[Union[str, Provider]] = None
+    ):
+        from kloppy.config import get_config
+
+        if not to_coordinate_system:
+            to_coordinate_system = get_config("coordinate_system")
+
+        if not to_coordinate_system:
+            to_coordinate_system = Provider.KLOPPY
+
+        to_dataset_type = None
+        if isinstance(to_coordinate_system, str):
+            if ":" in to_coordinate_system:
+                provider_name, dataset_type_name = to_coordinate_system.split(
+                    ":"
+                )
+                to_coordinate_system = Provider[provider_name.upper()]
+                to_dataset_type = DatasetType[dataset_type_name.upper()]
+            else:
+                to_coordinate_system = Provider[to_coordinate_system.upper()]
+
+        self.to_coordinate_system = to_coordinate_system
+        self.to_dataset_type = to_dataset_type
+
+    def build(
+        self,
+        length: float,
+        width: float,
+        provider: Provider,
+        dataset_type: DatasetType,
+    ):
+        from_coordinate_system = build_coordinate_system(
+            provider, length=length, width=width, dataset_type=dataset_type
```

Comment:
Perhaps use the same formatting as for the `to_coordinate_system`? The similarties and differences between the two calls to `build_coordinate_system` would become more apparent.